File: slibai/backend/app/schemas/tool.py
from pydantic import BaseModel
from typing import Optional


class AITool(BaseModel):
    id: int
    name: str
    category: str
    function: str
    description: str
    developer: Optional[str] = ""
    version: Optional[str] = ""
    cost: Optional[str] = ""
    compatibility: Optional[str] = ""
    dependencies: Optional[str] = ""
    social_impact: Optional[str] = ""
    example_code: Optional[str] = ""
    official_url: Optional[str] = ""

File: slibai/frontend/src/types/tool.ts
export interface AITool {
  id: number
  name: string
  category: string
  function: string
  description: string
  developer?: string
  version?: string
  cost?: string
  compatibility?: string
  dependencies?: string
  social_impact?: string
  example_code?: string
  official_url?: string
}

export interface CategoryStat {
  category: string
  count: number
}

export interface SearchResponse {
  results: AITool[]
  detected_category: string | null
  total_results: number
  query: string
}
